### PR TITLE
Fix the build on Rocky8

### PR DESCRIPTION
**Pull request by geffrak**
### Summarize your change.

Uses the system OpenSSL on RHEL8 while building OpenRV.

### Describe the reason for the change.

RHEL8 uses the same version of OpenSSL than the current OpenRV, and they enter in conflict during the build. There were a bit of documentation about how to avoid the conflict, but this is implementing a real fix.

### Describe what you have tested and on which operating system.

Tested on Rocky8, and on Autodesk's RV CI to make sure nothing breaks with this change.

### Add a list of changes, and note any that might need special attention during the review.

With this change, we no longer package OpenSSL in RV if building on a Rocky8. CentOS7 provides a openssl11 package that gives OpenSSL 1.1.1 and Rocky9 provides a compat-openssl package.

### If possible, provide screenshots.

N/A

### CODE REVIEW

File: src/lib/files/WFObj
Comment (geffrak) [inline src/lib/files/WFObj:1]: This is for the .gitignore bit. It wasn't working on case sensitive file systems.

File: cmake/dependencies/python3.cmake
Comment (geffrak) [inline cmake/dependencies/python3.cmake:1]: Only specify openssl-dir if homemade

File: docs/build_system/config_linux_rocky8.md
Comment (geffrak) [inline docs/build_system/config_linux_rocky8.md:1]: No need for this hack anymore.

File: src/build/make_pyside.py
Comment (geffrak) [inline src/build/make_pyside.py:1]: make OPENSSL_OUTPUT_DIR optional

**Review by bernie-laberge — APPROVED** [review]
Superb !

**Review by rogernelson — APPROVED** [review]

**Pull request merged**